src/ripple/rpc/handlers/AccountOffers.cpp

```diff
@@ -24,6 +24,19 @@
 
 namespace ripple {
 
+void appendOfferJson (std::shared_ptr<SLE const> const& offer,
```

Comment:
There is such a thing as over-encapsulation.  Sometimes it is important for the client to know that it is holding a `shared_ptr` as opposed to a `unique_ptr` or raw pointer.  By spelling it out explicitly, one no longer has to go look up the pointer type in some other header.  The result is more verbose, but also more readable.

That's not to say that nested type aliases are never a good thing.  It is a judgment call that should be made on a case by case basis.